test/parallel/test-fs-watch-recursive-linux-promise.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const common = require('../common');
+
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+if (common.isIBMi)
+  common.skip('IBMi does not support `fs.watch()`');
+
+const { randomUUID } = require('crypto');
+const assert = require('assert');
+const path = require('path');
+const fs = require('fs/promises');
+const fsSync = require('fs');
+
+const tmpdir = require('../common/tmpdir');
+const testDir = tmpdir.path;
+tmpdir.refresh();
+
+(async function run() {
+  // Add a file to already watching folder
+
+  const testsubdir = await fs.mkdtemp(testDir + path.sep);
+  const file = `${randomUUID()}.txt`;
+  const filePath = path.join(testsubdir, file);
+  const watcher = fs.watch(testsubdir, { recursive: true });
+
+  let interval;
+
+  process.nextTick(common.mustCall(() => {
+    interval = setInterval(() => {
+      fsSync.writeFileSync(filePath, 'world');
+    }, 500);
+  }));
+
+  try {
+    for await (const payload of watcher) {
+      const { eventType, filename } = payload;
+
+      assert.ok(eventType === 'change' || eventType === 'rename');
+
+      if (filename === file) {
+        clearInterval(interval);
+        interval = null;
+        break;
+      }
+    }
+  } catch (error) {
+    if (error.name !== 'AbortError') {
+      assert.fail(error);
+    }
```

Comment:
Why would there be an `AbortError` here? And why would it be OK to ignore it?

Reply:
breaking a `for await` of Readable stream throws an `AbortError`.

Reply:
That seems like a problematic DX, it looks like the macOS version doesn't have this issue.

Reply:
macOS version also has this. Here's the gist and the log with it: https://gist.github.com/anonrig/60ae77cd3b59fdf0069385cca103bff2

Reply:
My main goal is to keep https://ci.nodejs.org/job/node-test-commit-ibmi/ at least unstable (yellow) and not to introduce new test failures.
We run that CI daily but not on pull requests because unfortunately it is very slow. Feel free to start runs of that job through Jenkins if you need to check anything on IBM i.

Reply:
That's true. `break` does not throw AbortError but using watcher with AbortController throws error. Since watcher uses Readable stream for async iteration, and streams come with pre-defined abort signals: break triggers the abort controller to abort, which throws the actual error. 

I don't know an easy way without altering the `streams` to solve this issue. I think this is an issue with the streams, not with this implementation. Wdyt @aduh95?

Reply:
Solved this.

Reply:
I think we're missing something here, I don't think the issue is with the streams API. If I run the following code:

```js
for await (const line of process.stdin) break;

console.log('no AbortError');
```

I see no AbortError. I think it has to do with the way we use this API.